packages/indexer/lib/index.js

```diff
@@ -190,88 +191,90 @@ class CellCollector {
 class TransactionCollector {
   constructor(
     indexer,
-    { lock = null, type = null } = {},
+    { input_lock = null, output_lock = null, input_type = null, output_type } = {},
```

Comment:
We need to also keep the original `lock` and `type` query, for compatibility reasons.

Reply:
In stead of keepping both `lock | type | input_lock | output_lock | input_type | output_type` to enable a fine-grained query, I would propose to add new field in existing `lock | type` : 

```
lock: {
  script: {
    code_hash:
        "0x0000000000000000000000000000000000000000000000000000000000000000",
      hash_type: "data",
      args: "0x62e907b15cbf27d5425399ebf6f0fb50ebb88f18",
  },
 io_type: null // or 0 - input, 1 - output
}
```

It seems like a more natural way to support #16

Reply:
2 things:

1. We cannot break compatibility here, so if you do this path, lock should accept both a plain script, or an object containing script and `io_type` like shown above.
2. `io_type` is an internal detail, we shouldn't expose the values out here, it's better to use `"input"` and `"output"`, which will have better typing support at TypeScript side.